easy-ui-react/src/DropdownButton/DropdownButton.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { ReactNode, useRef } from "react";
+import ExpandMoreIcon from "@easypost/easy-ui-icons/ExpandMore";
+import { useButton } from "react-aria";
+import { Icon } from "../Icon";
+import { ButtonColor } from "../types";
+import { classNames, variationName } from "../utilities/css";
+import { LogWarningIfInvalidColorVariantCombination } from "../utilities/button";
+
+import styles from "./DropdownButton.module.scss";
+
+export type DropdownButtonVariant = "filled" | "outlined";
+
+export type DropdownButtonProps = {
+  /** Button color */
+  color?: ButtonColor;
+  /** Button variant */
+  variant?: DropdownButtonVariant;
+  /** Disables button */
+  isDisabled?: boolean;
+  /** Content inside button  */
+  children?: ReactNode;
+};
+
+export function DropdownButton(props: DropdownButtonProps) {
+  const {
+    color = "primary",
+    variant = "filled",
+    isDisabled = false,
+    children = "Button",
+  } = props;
+
+  const ref = useRef(null);
+  const { buttonProps } = useButton(props, ref);
+
+  LogWarningIfInvalidColorVariantCombination(color, variant);
+
+  return (
+    <button
+      disabled={isDisabled}
+      ref={ref}
+      className={classNames(
+        styles.DropdownButton,
+        styles[variationName("color", color)],
+        styles[variationName("variant", variant)],
+      )}
+      {...buttonProps}
+    >
+      <span>{children}</span>
+      <div className={classNames(styles.pipeSeparator)}></div>
+      <Icon symbol={ExpandMoreIcon} size="sm" />
```

Comment:
i went ahead and removed the size prop and so it defaults to md just like the icons on the regular button. i think i misinterpreted the figma and it looks like the md size aligns a bit more with what we want

Reply:
the icon looks like the right size (apart from it being a different weight, which we could configure now if we need to by adding the `@material-symbols/svg-400` package and creating an `Add400` icon in `easy-ui-icons` but that's up to you and design as to whether or not we feel we need it) but looks like now the spacing is too wide to the left and right of the icon:

<img width="172" alt="image" src="https://user-images.githubusercontent.com/752942/226396213-63697ccb-a89b-4923-a324-a74810ca4d84.png">

<img width="155" alt="image" src="https://user-images.githubusercontent.com/752942/226396368-33f6c740-8b42-40bc-b3e7-e96a3916b8b0.png">

this is because the icon itself has a `24px` viewbox:

<img width="299" alt="image" src="https://user-images.githubusercontent.com/752942/226396964-46eb454b-e7d5-4dd1-a9c5-e86ff0a148ed.png">

ideally, in Figma this would be represented with a `24px` view box around the icon so that we know the exact spacing around it. otherwise we will need to do manual calculations, but i wouldn't recommend that. we should talk to design about this. in general, `md` icons create a `24px` view box, `sm` icons create a `16px` view box, `lg` icons create a `32px` view box. these are settings directly from Material Symbols itself

Reply:
added the 400 variant and removed the previous one, but let me know if you prefer having the other one as well and ill update the name to this one as ExpandMore400.

i brought it to design's attention that the icon has a 24px viewbox thats coming from Material and with the original padding as implemented in the specs, its causing the net space around the icon (and the overall button) to be wider. i asked design if they felt it was too wide and if so, how should we adjust the space. the suggestion was to reduce the spacing from the text and the pipe to 12px as well as reducing the spacing between the pipe and icon to 12px.

Reply:
> added the 400 variant and removed the previous one, but let me know if you prefer having the other one as well and ill update the name to this one as ExpandMore400

i like that without the 400 in the name the consumer doesn't necessarily have to think about weight at all. that said, and you can let me know your thoughts about this, but my only concern with the name is that if we generally use 300 as the icon weight, will it be confusing if this particular icon is 400? and if for whatever we reason we need to add ExpandMore 300 weight in the future, will it be confusing to add that as ExpandMore300, and the consumer would assume that the default instead is 300?

> i brought it to design's attention that the icon has a 24px viewbox thats coming from Material and with the original padding as implemented in the specs, its causing the net space around the icon (and the overall button) to be wider. i asked design if they felt it was too wide and if so, how should we adjust the space. the suggestion was to reduce the spacing from the text and the pipe to 12px as well as reducing the spacing between the pipe and icon to 12px. although that still does leave the spacing unbalanced if it was the original intention to balance them

got ya. i strongly discourage having our components differ from what is in Figma. future developers and designers will be confused as to why there's a difference and it will create problems that cascade to other areas of the design that use this button. i believe we need to reconcile the differences to ensure they're precisely the same

Reply:
> i like that without the 400 in the name the consumer doesn't necessarily have to think about weight at all. that said, and you can let me know your thoughts about this, but my only concern with the name is that if we generally use 300 as the icon weight, will it be confusing if this particular icon is 400? and if for whatever we reason we need to add ExpandMore 300 weight in the future, will it be confusing to add that as ExpandMore300, and the consumer would assume that the default instead is 300?

i do like the approach that the consumer doesn’t necessarily need to think about weight, but as you alluded to, we may have to carry around more than one variation and in effect give the assumption that 300 is the default. hmm, do you anticipate that we will need to carry around multiple weights? i suppose thats more of a design question. im wondering if it would be good to at the weight suffix to every icon, although i do think that has the potential to slightly worsen the DX

> got ya. i strongly discourage having our components differ from what is in Figma. future developers and designers will be confused as to why there's a difference and it will create problems that cascade to other areas of the design that use this button. i believe we need to reconcile the differences to ensure they're precisely the same

i agree, i requested for design to add the 24px viewbox/bounding box in the figma so we have some documentation for the actual implementation

Reply:
based on updated specs, there isn't much else to change on the dev side i dont think, but as of now the padding calculations are handled in the variant selectors. this wont work for the `IconButton` but i think it makes sense to handle it on the dedicated PR for that one. 

i decided to include the `400` in the name and updated the readme accordingly. i think its sensible given that the rest are (for now) at a default of 300, but if you think this does worsen the DX, im ok with removing it until we come up with a cleaner way to handle it